lib/arel/visitors/oracle_common.rb

```diff
@@ -9,7 +9,9 @@ module OracleCommon
         # Fixes ORA-00932: inconsistent datatypes: expected - got CLOB
         def visit_Arel_Nodes_Equality(o, collector)
           left = o.left
+
           return super unless %i(text binary).include?(cached_column_for(left)&.type)
+          return super if o.right.nil?
```

Comment:
Just a nitpick, maybe create a variable `right` and check it for being `nil` earlier because it seems a more lightweight check than the other one. But it's fine as it is too.